src/telemetry.cpp

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "telemetry.h"
+#include "leakdetector.h"
+#include "logger.h"
+#include "mozillavpn.h"
+
+#include "../../glean/telemetry/gleansample.h"
+
+// 10 seconds of connection.
+constexpr int CONNECTION_STABILITY_MSEC = 10000;
```

Comment:
As a minor improvement, I would recommend increasing this to 45-60 seconds. The ping statistics are averaged over a 30-second window, and this will ensure that any chaos from the connection change will fall out of that window and won't tarnish our statistics.